### PR TITLE
Task_9975_9989: BookList and BookShelf Screens

### PR DESCRIPTION
**Pull request by khanhnn-1791**
### Issue Link :link:

https://edu-redmine.sun-asterisk.vn/issues/9975
https://edu-redmine.sun-asterisk.vn/issues/9989
### Goals :soccer:

https://marvelapp.com/project/3891276/screen/55142749
Create UI for BookShelf Screen
+ Cancel, Done label
+ Navigation bar title
+ 3 processing options list using tableview
+ Each tableviewcell includes: progress label and a button
+ Remove button at the bottom view
Create a UI action
- Click close button to go to the previous view
- Click done button to save progress and go to the previous view
- Click cell to select current progress
- Click remove button to put book's status to ""Want to read""
Fetch data from API and display to the UI
https://marvelapp.com/project/3891276/screen/55157454
Create UI for BookList Screen
+ Navigation Bar title
+ Back Label
+ Books list using tableview
+ Book Item using tableviewcell
Create a UI action
- Click book to go to detail book
- Click back to go to the previous view
- Click book to go to detail book
### Implementation Details :construction:

https://gfycat.com/glitteringvariableantelopegroundsquirrel

### CODE REVIEW

**Comment by khanhnn-1791** [comment]
ready

File: ios-i-book-lover/Scenes/Cell/BookCell/BookListCell.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    @IBOutlet private weak var authorLabel: UILabel!
+    @IBOutlet private weak var ratingView: RatingView!
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+    }
+    
+    func setContent(book: Book) {
+        guard let author = book.author,
+            let url = URL(string: book.imageUrl)
```
Comment (tienpx3010) [inline ios-i-book-lover/Scenes/Cell/BookCell/BookListCell.swift]:
```suggestion
            let url = URL(string: book.imageUrl) else
```

File: ios-i-book-lover/Scenes/Cell/BookCell/BookListCell.swift [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        }
+        authorLabel.text = author.name
+        titleLabel.text = book.title
+        ratingView.setRating(rate: Double(book.averageRating) ?? 0.0)
+        let resource = ImageResource(downloadURL: url, cacheKey: book.imageUrl)
+        bookImageView.kf.setImage(with: resource)
+    }
+}
+
+extension BookListCell: NibReusable {
```
Comment (tienpx3010) [inline ios-i-book-lover/Scenes/Cell/BookCell/BookListCell.swift]: UITableViewCell, NibReusable

File: ios-i-book-lover/Scenes/Home/BookDetail/BookDetailViewController.swift [hunk 3]
```diff
@@ -78,6 +82,50 @@ final class BookDetailViewController: ButtonBarPagerTabStripViewController {
...
+            wantToReadView.do {
+                $0.layer.backgroundColor = UIColor.white.cgColor
+                $0.layer.borderColor = UIColor.mainColor.cgColor
+                $0.layer.borderWidth = 1
+            }
+            wantToReadLabel.do {
+                $0.textColor = .mainColor
+                $0.text = Constants.progresses[progress]
+            }
+            wantToReadImageView.do {
```
Comment (tienpx3010) [inline ios-i-book-lover/Scenes/Home/BookDetail/BookDetailViewController.swift]:
```suggestion
            wantToReadImageView.image = UIImage(named: "ic_down_2")
```

File: ios-i-book-lover/Scenes/Home/BookShelfViewController.swift [hunk 4]
```diff
@@ -8,11 +8,194 @@
...
+                try managedContext.save()
+            } catch {
+                print(error)
+            }
+        } catch {
+            print(error)
+        }
+    }
+    
+    private func remove() {
```
Comment (tienpx3010) [inline ios-i-book-lover/Scenes/Home/BookShelfViewController.swift]: remove what?

**Review by tienpx3010 — APPROVED** [review]

**Pull request merged**